### ISSUE TITLE
False positives with CSS Custom Properties in selector-type-no-unknown

### ISSUE THREAD

**Issue by kevinSuttle** [stylelint/stylelint#1791]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Bug where `selector-type-no-unknown` doesn't understand CSS Custom Property syntax by default. 

> Which rule, if any, is this issue related to?

`selector-type-no-unknown`

> What CSS is needed to reproduce this issue?

```css
:root {
  --ThemeAnimationStyles {
    transition-duration: .2s;
    transition-timing-function: cubic-bezier(.075, .82, .165, 1);
  }
}
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "extends": "stylelint-config-standard",
  "rules": {
    "number-leading-zero": "never",
    "custom-property-no-outside-root": true,
    "root-no-standard-properties": true,
    "selector-root-no-composition": true,
    "declaration-block-no-duplicate-properties": true,
    "no-eol-whitespace": true
  }
}
```

> Which version of stylelint are you using? 

`7.1.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Maybe? 

> What did you expect to happen?

"No warnings to be flagged."

> What actually happened (e.g. what warnings or errors you are getting)?

e.g. "The following warnings were flagged:"

```shell
test.css
✖  Unexpected unknown type selector "--ThemeAnimationStyles"
```

> Workaround 

```json
"selector-type-no-unknown": [true, {
    	"ignoreTypes": ["/^--/", "--Theme"]
  }]
```

**Comment by evilebottnawi** [comment]
@kevinSuttle by default we should ignore this, why we should allow CSS Custom Properties?

**Comment by jeddy3** [comment]
@evilebottnawi I think this issue is that custom property sets (e.g.  `--ThemeAnimationStyles`) are being misclassified as *type selectors* in the `selector-type-no-unknown` rule.

I think we need a `isCustomPropertySet` util (like our preexisting `isCustomProperty` util), which this rule can use to ignore custom property sets outright.

**Comment by evilebottnawi** [comment]
@jeddy3 agree

**Comment by davidtheclark** [comment]
I'm a little confused here. Is this syntax in a spec?

```css
--ThemeAnimationStyles {
  transition-duration: .2s;
  transition-timing-function: cubic-bezier(.075, .82, .165, 1);
}
```

**Comment by evilebottnawi** [comment]
@davidtheclark not correct, not found in spec https://drafts.csswg.org/css-variables/
@kevinSuttle what are you using for this construction? `postcss` plugin?

**Comment by kevinSuttle** [comment]
https://github.com/postcss/postcss-custom-properties - which is to W3C spec as much as it can be.

**Comment by davidtheclark** [comment]
@kevinSuttle Quickly scanning the postcss-custom-properties README and the W3C spec draft it's trying to support, I do not see anything suggesting that your syntax is valid — that is, I don't see anything suggesting a custom property can define a *block*: it's always a single *declaration*, right? Or can you link to something that shows the block is ok?

**Comment by kevinSuttle** [comment]
Sorry, meant to post this one: https://github.com/pascalduez/postcss-apply

**Comment by davidtheclark** [comment]
@kevinSuttle According to the README of that it looks like you should have a colon in your declaration. What happens if you add one?
